File: Spotter/UI/spotter_desktop/lib/core/models/ticket_response.dart
class TicketResponse {
  final int id;
  final int userId;
  final String userFullName;
  final int eventId;
  final String eventTitle;
  final String ticketTypeName;
  final String typeName;
  final String qrCodePayload;
  final int status;
  final String statusName;
  final DateTime issuedAt;
  final DateTime? usedAt;

  TicketResponse({
    required this.id,
    required this.userId,
    required this.userFullName,
    required this.eventId,
    required this.eventTitle,
    required this.ticketTypeName,
    required this.typeName,
    required this.qrCodePayload,
    required this.status,
    required this.statusName,
    required this.issuedAt,
    this.usedAt,
  });

  factory TicketResponse.fromJson(Map<String, dynamic> json) {
    return TicketResponse(
      id: json['id'] as int,
      userId: json['userId'] as int,
      userFullName: json['userFullName'] as String,
      eventId: json['eventId'] as int,
      eventTitle: json['eventTitle'] as String,
      ticketTypeName: json['ticketTypeName'] as String,
      typeName: json['typeName'] as String,
      qrCodePayload: json['qrCodePayload'] as String,
      status: json['status'] as int,
      statusName: json['statusName'] as String,
      issuedAt: DateTime.parse(json['issuedAt'] as String),
      usedAt: json['usedAt'] != null
          ? DateTime.parse(json['usedAt'] as String)
          : null,
    );
  }
}

File: Spotter.Model/Responses/TicketResponse.cs
using Spotter.Model.Enums;

namespace Spotter.Model.Responses
{
    public class TicketResponse
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string UserFullName { get; set; } = string.Empty;
        public int EventId { get; set; }
        public string EventTitle { get; set; } = string.Empty;
        public DateTime? EventStartsAt { get; set; }
        public string VenueName { get; set; } = string.Empty;
        public string CityName { get; set; } = string.Empty;
        public string TicketTypeName { get; set; } = string.Empty;
        public string TypeName { get; set; } = string.Empty;
        public decimal Price { get; set; }
        public string QrCodePayload { get; set; } = string.Empty;
        public TicketStatus Status { get; set; }
        public string StatusName { get; set; } = string.Empty;
        public DateTime IssuedAt { get; set; }
        public DateTime? UsedAt { get; set; }
    }
}
